server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -230,9 +252,13 @@ private Flags getMessageFlags(CreationMessage message) {
         return result;
     }
 
-    private void sendMessage(MailboxMessage<Id> mailboxMessage, Message jmapMessage, MailboxSession session) throws MessagingException, IOException {
-        Mail mail = mailFactory.build(mailboxMessage, jmapMessage);
-        MailMetadata metadata = new MailMetadata(jmapMessage.getId(), session.getUser().getUserName());
-        mailSpool.send(mail, metadata);
+    private void sendMessage(MailboxMessage<Id> mailboxMessage, Message jmapMessage, MailboxSession session) throws MessagingException {
+        try {
+            Mail mail = mailFactory.build(mailboxMessage, jmapMessage);
+            MailMetadata metadata = new MailMetadata(jmapMessage.getId(), session.getUser().getUserName());
+            mailSpool.send(mail, metadata);
+        } catch (IOException e) {
+            Throwables.propagate(e);
```

Comment:
So a sending failut and subsequent request that may not use mailSpool (eg : upcoming saving to draft) will fail ?